### PR TITLE
Allow pathlib.Path as Slide path paramerer

### LINKED ISSUE #225: Slide accepting pathlib.Path as path of slide on disk
OpenSlide only supports `str` as for path slides, but on our side we could accept `pathlib.Path` as well and convert it to `str` internally.

### PR DESCRIPTION
**Pull request by alessiamarcolini**
fixes #225

### CODE REVIEW

File: tests/unit/test_tiler.py [hunk 1]
```diff
@@ -340,12 +341,27 @@ def it_can_generate_random_tiles_even_when_coords_are_not_valid(
...
+            (str),
+            (Path),
+        ],
+    )
+    def it_can_extract_random_tiles(
+        self,
+        request,
+        tmpdir,
+        path_type_transform,
+    ):
```
Comment (ernestoarbitrio) [inline tests/unit/test_tiler.py:356]: I don't see the need to have both cases here. Considering that this change is a `Slide` change and in here we're testing the tiler so this parametrisation is pointless IMHO. You already have a test with both str and Path in the slides unit tests ... so I'd revert the change here

File: src/histolab/slide.py [hunk 2]
```diff
@@ -52,13 +52,18 @@ class Slide:
...
         Path where thumbnails and scaled images will be saved to.
     """
 
-    def __init__(self, path: str, processed_path: str) -> None:
+    def __init__(
+        self, path: Union[str, pathlib.Path], processed_path: Union[str, pathlib.Path]
+    ) -> None:
+        if isinstance(path, pathlib.Path):
+            path = str(path)
+
```
Comment (ernestoarbitrio) [inline src/histolab/slide.py:66]: I think this if is not needed. just cast the path to string in any case and we're done 🤷🏽‍♂️

File: src/histolab/slide.py [hunk 3]
```diff
@@ -52,13 +52,18 @@ class Slide:
...
         self._path = path
```
Comment (ernestoarbitrio) [inline src/histolab/slide.py:67]:
```suggestion
        self._path = str(path)
```

**Review by ernestoarbitrio — CHANGES_REQUESTED** [review]
Some comments left

File: src/histolab/slide.py (hunk 2 above)
Comment (alessiamarcolini) [reply src/histolab/slide.py:66]: ehm nope
```python
In [1]: str(None)
Out[1]: 'None'
```

File: tests/unit/test_tiler.py (hunk 1 above)
Comment (alessiamarcolini) [reply tests/unit/test_tiler.py:356]: > You already have a test with both str and Path in the slides unit tests

yes but since we are lazy loading the slide, that test is not meaningful if the goal is checking if histolab can handle `pathlib.Path` before opening a slide with openslide

I'm adding another unit test that opens a slide from disk to check the `._wsi` attribute and I'm reverting this change

File: src/histolab/slide.py (hunk 2 above)
Comment (ernestoarbitrio) [reply src/histolab/slide.py:66]: ok, seems fair so ...why we don't use a ternary operator in the assigment?
```python
self._path = str(path) if isinstance(path, pathlib.Path) else path
```

**Review by ernestoarbitrio — CHANGES_REQUESTED** [review]
just a further minor comment

File: src/histolab/slide.py (hunk 2 above)
Comment (alessiamarcolini) [reply src/histolab/slide.py:66]: yeah why not

**Review by ernestoarbitrio — APPROVED** [review]
Lgtm
